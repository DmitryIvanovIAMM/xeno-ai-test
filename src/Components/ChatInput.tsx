'use client';

import { useState } from 'react';
import { useQueryAiHook } from '@/hooks/useQueryAiHook';

const ChatInput = () => {
  const { chatAiState, sendAIQuery } = useQueryAiHook();

  return (
    <div>
      <input type='text' />
      <button disabled={chatAiState.isQuerying} onClick={sendAIQuery}>
        Send
      </button>
    </div>
  );
};
export default ChatInput;
