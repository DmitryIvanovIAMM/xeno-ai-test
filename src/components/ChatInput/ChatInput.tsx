'use client';

import React from 'react';
import { useQueryAiHook } from '@/hooks/useQueryAiHook';
import { Button, InputAdornment, TextField } from '@mui/material';
import SendIcon from '@mui/icons-material/Send';
import styles from './chat-input.module.css';
import { AccountCircle } from '@mui/icons-material';

const ChatInput = () => {
  const { chatAiState, sendAIQuery, setInputValue } = useQueryAiHook();

  const keyPressed = (event: React.KeyboardEvent<HTMLInputElement>) => {
    if (event.keyCode == 13) {
      sendAIQuery();
    }
  };

  return (
    <div className={styles.container}>
      <TextField
        id='outlined-multiline-static'
        label='Put question here: '
        fullWidth={true}
        value={chatAiState.currentInput}
        margin='dense'
        size='small'
        onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
          setInputValue(event.target.value);
        }}
        onKeyUp={keyPressed}
        disabled={chatAiState.isQuerying}
        slotProps={{
          input: {
            startAdornment: (
              <InputAdornment position='start' style={{ marginLeft: 0 }}>
                <AccountCircle />
              </InputAdornment>
            ),
          },
        }}
      />
      <div className={styles.button}>
        <Button
          variant='contained'
          onClick={sendAIQuery}
          disabled={chatAiState.isQuerying || chatAiState.currentInput.length === 0}
          style={{ height: '42px', fontSize: '12px' }}
          endIcon={<SendIcon />}
        >
          Send
        </Button>
      </div>
    </div>
  );
};
export default ChatInput;
