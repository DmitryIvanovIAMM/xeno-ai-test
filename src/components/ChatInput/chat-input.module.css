.container {
    margin: 10px 10px 10px 10px;
    display: flex;
    flex-direction: row;
}
.button {
    margin: 6px 0px 10px 10px;
    text-align: center;
}