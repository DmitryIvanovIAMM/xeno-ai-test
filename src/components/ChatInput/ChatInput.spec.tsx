import { render } from '@testing-library/react';
import ChatInput from '@/components/ChatInput/ChatInput';

const setup = async () => {
  const container = render(<ChatInput />);
  return {
    container,
  };
};

describe('ChatInput component', () => {
  it('should render correctly', async () => {
    const { container } = await setup();

    expect(container).toMatchSnapshot();
  });

  it('should contain "Send query" button', async () => {
    const { container } = await setup();

    container.getByRole('button', {
      name: 'Send',
    });
  });
});
