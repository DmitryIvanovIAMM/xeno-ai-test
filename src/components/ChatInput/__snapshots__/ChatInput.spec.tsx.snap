// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInput component should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-xanwcj-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="outlined-multiline-static"
            id="outlined-multiline-static-label"
          >
            Put question here: 
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-multiline-static"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-81qg8w"
              >
                <span>
                  Put question here: 
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="button"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
            disabled=""
            style="height: 58px;"
            tabindex="-1"
            type="button"
          >
            Send query
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-xanwcj-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="outlined-multiline-static"
          id="outlined-multiline-static-label"
        >
          Put question here: 
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-multiline-static"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-81qg8w"
            >
              <span>
                Put question here: 
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="button"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
          disabled=""
          style="height: 58px;"
          tabindex="-1"
          type="button"
        >
          Send query
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
